android-lib/src/main/java/com/kinvey/android/store/DataStore.java

```diff
@@ -682,11 +676,7 @@ public void pull(Query query, KinveyPullCallback<T> callback) {
     public void pull(KinveyPullCallback<T> callback) {
```

Comment:
Wouldn't it be better for this method to call the other method `pull (query, callback)` with a `null` query? Just to avoid repeated checking of preconditions.

Reply:
Yeah makes sense.  I was just reverting this back to the way it was, but I'll change it to consolidate the logic.